File: Aigilas/Aigilas/Management/InputInitializer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SPX.Core;
using Microsoft.Xna.Framework.Input;

namespace Agilas.Management
{
    public class Commands
    {
        public const int MoveUp = 0;
        public const int MoveDown = 1;
        public const int MoveLeft = 2;
        public const int MoveRight = 3;
        public const int Confirm = 4;
        public const int Inventory = 5;
        public const int Skill = 6;
        public const int CycleLeft = 7;
        public const int CycleRight = 8;
        public const int Cancel = 9;
        public const int Start = 10;
        public const int Back = 11;
        public const int LockSkill = 12;
        public const int HotSkill1 = 13;
        public const int HotSkill2 = 14;
        public const int HotSkill3 = 15;
    }
    public class InputInitializer:IInputInitializer
    {

        private CommandDefinition Make(int command, Keys key, Buttons button, int lockContext)
        {
            return new CommandDefinition(command, key, button, lockContext);
        }

        public ICollection<CommandDefinition> GetCommands()
        {
            return new List<CommandDefinition>()
            {
                Make(Commands.MoveUp,Keys.Up,Buttons.LeftThumbstickUp,Contexts.Nonfree),
                Make(Commands.MoveDown,Keys.Down,Buttons.LeftThumbstickDown,Contexts.Nonfree),
                Make(Commands.MoveLeft,Keys.Left,Buttons.LeftThumbstickLeft,Contexts.Nonfree),
                Make(Commands.MoveRight,Keys.Right,Buttons.LeftThumbstickRight,Contexts.Nonfree),
                Make(Commands.Confirm,Keys.Space,Buttons.RightTrigger,Contexts.All),
                Make(Commands.Inventory,Keys.E,Buttons.DPadUp,Contexts.All),
                Make(Commands.Cancel,Keys.R,Buttons.X,Contexts.All),
                Make(Commands.Start,Keys.Enter,Buttons.Start,Contexts.All),
                Make(Commands.Back,Keys.Back,Buttons.Back,Contexts.All),
                Make(Commands.CycleRight,Keys.D,Buttons.RightShoulder,Contexts.All),
                Make(Commands.CycleLeft,Keys.A,Buttons.LeftShoulder,Contexts.All),
                Make(Commands.Skill,Keys.S,Buttons.A,Contexts.All),
                Make(Commands.LockSkill,Keys.RightControl,Buttons.LeftTrigger,Contexts.All),
                Make(Commands.HotSkill1,Keys.D1,Buttons.X,Contexts.All),
                Make(Commands.HotSkill2,Keys.D2,Buttons.Y,Contexts.All),
                Make(Commands.HotSkill3,Keys.D3,Buttons.B,Contexts.All)
            };
        }
    }
}

File: proj/src/aigilas/management/InputInitializer.java
package aigilas.management;

import spx.io.*;

import java.util.ArrayList;
import java.util.List;

public class InputInitializer implements IInputInitializer {
    private CommandDefinition make(Commands command, Keys key, Buttons button, Contexts lockContext) {
        return new CommandDefinition(command, key, button, lockContext);
    }

    @Override
    public List<CommandDefinition> getCommands() {
        ArrayList<CommandDefinition> result = new ArrayList<CommandDefinition>();
        result.add(make(Commands.MoveUp, Keys.Up, Buttons.LeftThumbstickUp, Contexts.Nonfree));
        result.add(make(Commands.MoveDown, Keys.Down, Buttons.LeftThumbstickDown, Contexts.Nonfree));
        result.add(make(Commands.MoveLeft, Keys.Left, Buttons.LeftThumbstickLeft, Contexts.Nonfree));
        result.add(make(Commands.MoveRight, Keys.Right, Buttons.LeftThumbstickRight, Contexts.Nonfree));
        result.add(make(Commands.Confirm, Keys.Space, Buttons.RightTrigger, Contexts.All));
        result.add(make(Commands.Inventory, Keys.E, Buttons.DPadUp, Contexts.All));
        result.add(make(Commands.Cancel, Keys.R, Buttons.X, Contexts.All));
        result.add(make(Commands.Start, Keys.Enter, Buttons.Start, Contexts.All));
        result.add(make(Commands.Back, Keys.Back, Buttons.Back, Contexts.All));
        result.add(make(Commands.CycleRight, Keys.D, Buttons.RightShoulder, Contexts.All));
        result.add(make(Commands.CycleLeft, Keys.A, Buttons.LeftShoulder, Contexts.All));
        result.add(make(Commands.Skill, Keys.S, Buttons.A, Contexts.All));
        result.add(make(Commands.LockSkill, Keys.RightControl, Buttons.LeftTrigger, Contexts.All));
        result.add(make(Commands.HotSkill1, Keys.D1, Buttons.X, Contexts.All));
        result.add(make(Commands.HotSkill2, Keys.D2, Buttons.Y, Contexts.All));
        result.add(make(Commands.HotSkill3, Keys.D3, Buttons.B, Contexts.All));
        result.add(make(Commands.ToggleDevConsole, Keys.OemTilde, Buttons.DPadDown, Contexts.All));
        return result;
    }
}
